src/d3a/models/strategy/market_maker_strategy.py

```diff
@@ -15,36 +15,63 @@
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
-from d3a_interface.read_user_profile import read_and_convert_identity_profile_to_float
+import d3a.constants
+from d3a.models.strategy.commercial_producer import CommercialStrategy
 from d3a_interface.constants_limits import GlobalConfig, ConstSettings
 from d3a_interface.device_validator import validate_market_maker
+from d3a_interface.read_user_profile import InputProfileTypes, \
+    read_and_convert_identity_profile_to_float, convert_identity_profile_to_float
 from d3a_interface.utils import key_in_dict_and_not_none
-from d3a.models.strategy.commercial_producer import CommercialStrategy
 
 
 class MarketMakerStrategy(CommercialStrategy):
-    parameters = ('energy_rate_profile', 'energy_rate', 'grid_connected')
+    parameters = ("energy_rate_profile", "energy_rate", "grid_connected",
+                  "energy_rate_profile_uuid")
 
-    def __init__(self, energy_rate_profile=None, energy_rate=None, grid_connected=True):
+    def __init__(self, energy_rate_profile=None, energy_rate=None, grid_connected=True,
+                 energy_rate_profile_uuid=None):
+        super().__init__()
         validate_market_maker(grid_connected=grid_connected)
-        if energy_rate_profile is not None:
-            energy_rate = read_and_convert_identity_profile_to_float(energy_rate_profile)
+
+        self.energy_rate = None
+
+        if d3a.constants.CONNECT_TO_PROFILES_DB and energy_rate_profile_uuid:
+            self.energy_rate_profile = None
+            self.energy_rate_input = None
+            self.energy_rate_profile_uuid = energy_rate_profile_uuid
         else:
-            energy_rate = read_and_convert_identity_profile_to_float(
-                ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE
-                if energy_rate is None else energy_rate)
-        GlobalConfig.market_maker_rate = energy_rate
+            self.energy_rate_profile = energy_rate_profile
+            self.energy_rate_input = energy_rate
+            self.energy_rate_profile_uuid = None
+
         self._grid_connected = grid_connected
-        super().__init__(energy_rate)
+
+        self._read_or_rotate_rate_profiles()
+
+    def event_activate_price(self):
+        pass
+
+    def _read_or_rotate_rate_profiles(self):
```

Comment:
The more I see it in the code, the more I think that an empty implementation of this method should live in the BaseStrategy class in order to denote that it is needed for the implementation of any strategy.